js/services/api/base.js

```diff
@@ -3,47 +3,64 @@ import firebase from 'firebase';
 const FIRBASE_VALUE_TYPE = 'value';
 
 const get = (path) => {
+  // returns a promise of an actual object, not a datasnapshot
   return firebase.database()
     .ref(path)
     .once(FIRBASE_VALUE_TYPE)
-    .then((snapshot) => {
-      return snapshot.val();
+    .then((result) => {
+      return result.val();
     });
 };
 
 // to add data with the id created by firebase
 const push = (path, data) => {
   return firebase.database()
-  .ref(path)
-  .push(data);
-   // return new Id?
+    .ref(path)
+    .push(data);
+    // return new Id?
+};
+
+const pushKey = (path) => {
```

Comment:
This is a strange name that implies you're pushing a key. Maybe `pushToPath` would be better? The `key` part is problematic, I think it'd be cleaner to just return the whole object and let the client get the `key` property themselves.